packages/components/organisms/f-checkout/src/services/tests/addressService.test.js

```diff
@@ -272,4 +272,72 @@ describe('addressService', () => {
             });
         });
     });
+
+    describe('doesAddressInStorageAndFormMatch ::', () => {
+        describe('when the address fields are the same', () => {
+            it('should return true', () => {
+                // Arrange
+                const storageAddress = {
+                    Line1: 'Fleet Place House',
+                    Line2: 'Farringdon',
+                    City: 'London',
+                    PostalCode: 'EC4M 7RF'
+                };
+
+                const formAddress = {
+                    line1: 'Fleet Place House',
+                    line2: 'Farringdon',
+                    locality: 'London',
+                    postcode: 'EC4M 7RF'
+                };
+
+                // Act & Assert
+                expect(addressService.doesAddressInStorageAndFormMatch(storageAddress, formAddress)).toEqual(true);
+            });
+        });
+
+        describe('when the city field is different', () => {
+            it('should return false', () => {
+                // Arrange
+                const storageAddress = {
+                    Line1: 'Fleet Place House',
+                    Line2: 'Farringdon',
+                    City: 'Newcastle',
+                    PostalCode: 'EC4M 7RF'
+                };
+
+                const formAddress = {
+                    line1: 'Fleet Place House',
+                    line2: 'Farringdon',
+                    locality: 'London',
+                    postcode: 'EC4M 7RF'
+                };
+
+                // Act & Assert
+                expect(addressService.doesAddressInStorageAndFormMatch(storageAddress, formAddress)).toEqual(false);
+            });
+        });
+
+        describe('when the postcode is different', () => {
+            it('should return false', () => {
+                // Arrange
+                const storageAddress = {
+                    Line1: 'Fleet Place House',
+                    Line2: 'Farringdon',
+                    City: 'London',
+                    PostalCode: 'EC4M 7RF'
```

Comment:
What would happen if we removed the postcode space?

Reply:
It would see them as different, but as mentioned in the description, this wouldn't be a regression because in ConsumerWeb they just deep equals the objects. I could trim the fields and lowercase everything, but I don't want to make the function extremely convoluted.

Reply:
The API idea seems like a good point happy to approve